model/atmosphere/dycore/tests/dycore_stencil_tests/test_solve_tridiagonal_matrix_for_w_forward_sweep.py

```diff
@@ -7,108 +7,92 @@
 # SPDX-License-Identifier: BSD-3-Clause
 import gt4py.next as gtx
 import numpy as np
-from gt4py.next.program_processors.runners.gtfn import run_gtfn
+import pytest
 
 from icon4py.model.atmosphere.dycore.stencils.solve_tridiagonal_matrix_for_w_forward_sweep import (
     solve_tridiagonal_matrix_for_w_forward_sweep,
 )
 from icon4py.model.common import dimension as dims
-from icon4py.model.common.grid.simple import SimpleGrid
 from icon4py.model.common.type_alias import vpfloat, wpfloat
 from icon4py.model.common.utils.data_allocation import random_field
+from icon4py.model.testing.helpers import StencilTest
 
 
-def solve_tridiagonal_matrix_for_w_forward_sweep_numpy(
-    vwind_impl_wgt: np.array,
-    theta_v_ic: np.array,
-    ddqz_z_half: np.array,
-    z_alpha: np.array,
-    z_beta: np.array,
-    z_exner_expl: np.array,
-    z_w_expl: np.array,
-    z_q_ref: np.array,
-    w_ref: np.array,
-    dtime,
-    cpd,
-) -> tuple[np.array]:
-    z_q = np.copy(z_q_ref)
-    w = np.copy(w_ref)
-    vwind_impl_wgt = np.expand_dims(vwind_impl_wgt, axis=-1)
+class TestSolveTridiagonalMatrixForWForwardSweep(StencilTest):
+    PROGRAM = solve_tridiagonal_matrix_for_w_forward_sweep
+    OUTPUTS = ("w", "z_q")
 
-    z_gamma = dtime * cpd * vwind_impl_wgt * theta_v_ic / ddqz_z_half
-    z_a = np.zeros_like(z_gamma)
-    z_b = np.zeros_like(z_gamma)
-    z_c = np.zeros_like(z_gamma)
-    z_g = np.zeros_like(z_gamma)
+    @staticmethod
+    def reference(
+        grid,
+        vwind_impl_wgt: np.array,
```

Comment:
we use officially `np.ndarray`
```suggestion
        vwind_impl_wgt: np.ndarray,
```

Reply:
You are right. I searched online and found this:
> numpy.array is a function that returns a numpy.ndarray object.
> There is no object of type numpy.array.

I copied this type annotation from another place where it was wrong. I will search and replace in all files.

Reply:
Maybe better to only fix this file, in this PR, and do the search and replace in all files in a separate PR.

Reply:
Yes focus only on this. We have been fixing these data types everywhere but there are so mane tests that we always end up missing some... sorry about the fact that another one was wrong